packages/frontend/src/state/lp/hooks.ts

```diff
@@ -0,0 +1,219 @@
+import { nearestUsableTick, TickMath } from '@uniswap/v3-sdk'
+import { fromTokenAmount } from '@utils/calculations'
+import { useAtomValue } from 'jotai'
+import { addressesAtom, isWethToken0Atom } from '../positions/atoms'
+import BigNumber from 'bignumber.js'
+import { OSQUEETH_DECIMALS } from '@constants/index'
+import { controllerContractAtom, controllerHelperHelperContractAtom, nftManagerContractAtom, quoterContractAtom, squeethPoolContractAtom } from '../contracts/atoms'
+import useAppCallback from '@hooks/useAppCallback'
+import { addressAtom } from '../wallet/atoms'
+import { Contract } from 'web3-eth-contract'
+import { useHandleTransaction } from '../wallet/hooks'
+import { ethers } from 'ethers'
+import { useCallback } from 'react'
+import { useGetDebtAmount, useGetVault } from '../controller/hooks'
+
+/*** CONSTANTS ***/
+const COLLAT_RATIO_FLASHLOAN = 2
```

Comment:
we aren't using this in the code, the flashloan eth needs to be determined using this

Reply:
we're using the user input for this in an upcoming commit - may use const in future hooks